python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -562,9 +562,6 @@ def _overrideTransferMode(self, *datasets: Any, transfer: str | None = None) ->
 
     def _prepIngest(self, *datasets: FileDataset, transfer: str | None = None) -> _IngestPrepData:
         # Docstring inherited from Datastore._prepIngest.
-        if transfer is None:
```

Comment:
This check was probably added as a safeguard early on but then later other code was added.